app/views/recipes/show.html.erb

```diff
@@ -0,0 +1,70 @@
+<div class="recipe-show-page">
+  <div class="recipe-header">
+   <% if flash[:notice].present? %>
+    <div class="notice">
+      <%= flash[:notice] %>
+    </div>
+   <% end %>
+    <h2><%= @recipe.name %></h2>
+    <div class="top-line">
+      <span>Preparation Time: <%= @recipe.preparation_time %> minutes</span>
+      <span>
+      <%= form_for @recipe do |f| %>
+        <% if current_user == @recipe.user %>
+          <% if @recipe.public %>
+            <%= button_to  'Make Private ', recipe_path(@recipe), method: :patch, class: "btn btn-primary btn-sm" %>
+         <% else %>
+            <%= button_to  'Make public', recipe_path(@recipe), method: :patch, class: 'btn btn-primary btn-sm' %>
+          <% end %>
+        <% end %>
+        <% end %>
+      </span>
+    </div>
+  </div>
+  <p>Cooking Time: <%= @recipe.cooking_time %> minutes</p>
+  <p>Description: <%= @recipe.description %></p>
+  <div class="recipe-show-button-container">
+    <button class='btn btn-primary' id='gen-btn'>Generate Shopping List</button>
```

Comment:
- [ ] [REQUIRED] This button seems not to have a particular route and actions as it does not pop up the modal at the first click except when I refresh the page and I can see that a JS script is been implemented for it to work. Let's ensure this button displays the Generating shopping list modal after one click. The modal displays looks good 👍🏽